packages/es-dev-server/src/utils/utils.js

```diff
@@ -126,13 +126,19 @@ export async function isIndexHTMLResponse(ctx, appIndex) {
   }
 
   // if we're serving the app index, it's an index html response
-  if (ctx.url === appIndex) {
+  if (ctx.url === appIndex || ctx.url.includes('index.html')) {
```

Comment:
these kinds of assumptions keep biting us,  so I don't want to assume that `index.html` means a HTML file.

Reply:
If that's the case, do we need to add checking for the scenario where no `appIndex` is provided? The tests for `polyfills-loader` failed after adding the checking for accept type, as it is now giving out a false negative, unless we explicitly add `appIndex` in the tests' config.

Reply:
We will need to add the header in fetch requests in the tests probably.

The app index check is just a nice to have, but not strictly required.

Reply:
Updated the tests with the latest commit :)